src/Core/Framework/Plugin/Command/PluginListCommand.php

```diff
@@ -83,33 +124,63 @@ protected function execute(InputInterface $input, OutputInterface $output): int
                 $pluginUpgradeable ? 'Yes' : 'No',
             ];
 
+            ++$this->total;
+
             if ($pluginActive) {
-                ++$active;
+                ++$this->active;
             }
 
             if ($pluginInstalled) {
-                ++$installed;
+                ++$this->installed;
             }
 
             if ($pluginUpgradeable) {
-                ++$upgradeable;
+                ++$this->upgradeable;
             }
         }
 
+        if ($format === 'json') {
+            $this->outputJson($io, $pluginTable);
+        } elseif ($format === 'table') {
+            $this->outputTable($io, $pluginTable, $filter);
+        }
+
+        return 0;
+    }
+
+    protected function outputTable(
```

Comment:
please make the methods private

Reply:
Why? Making them protected leaves the possibility for a class that extends this class to change them if necessary.

Reply:
ok, so you already want to extend this? what do you want to change here in a plugin? are there other ways to make this happen?

Reply:
I don't want to extend it per se, I'm just saying that it annoys me to no end that another developer will close the door preemptively on me so I am not even *able* to extend it, might the need arise.

Reply:
I totally understand your point, but once this is in production code, this method belongs to the public API, which we will have to maintain. This makes it harder to refactor such methods, which must not necessarily be public/protected.

for now, you can leave it like this